ui/src/app/pages/signed-in/signed-in.tsx

```diff
@@ -105,22 +105,17 @@ export const SignedIn = (props: Props) => {
      *  See discussion on https://github.com/all-of-us/workbench/pull/4713
      */
     const checkStoresLoaded = async() => {
+      // AppComponent should be loading the server config.
       if (serverConfig.config) {
-        setServerConfigInitialized(true);
-        if (!profileStore.get().profile) {
+        if (!profile) {
```

Comment:
The flow will be a bit less natural, but this pattern may be more idiomatic for hooks:


```
useEffect(() => {
  if (!profile) {
    profileStore.get().load();
    return;
  }
  ...
}, [profileStore]);
```

This way it first enters with an unloaded profile, loads and stops. Then re-enters when the profileStore value has been updated and skips the conditional and runs the rest of the initialization routine. Same could be done with cdr Versions.

Either way - I think this is currently bugged because `profile` doesn't get updated to the latest value on L111, further lines would npe